### PR TITLE
smb: add unit tests

### PR DESCRIPTION
**Pull request by b1tg**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to redmine ticket: [4865](https://redmine.openinfosecfoundation.org/issues/4865)

I have begun adding unit tests into the SMB parsers, several of these tests have been completed, and I would like hearing your feedback on the code so far.

Signed-off-by: Lancer Cheng b1tg@protonmail.ch

### CODE REVIEW

**Review by inashivb — CHANGES_REQUESTED** [review]
Hello @b1tg !
Thank you for your contribution to our project. I have requested some minor changes. Your work is looking good to me. :)

Could you please add the source of test data to every test?

File: rust/src/smb/smb2_records.rs [hunk 1]
```diff
@@ -626,3 +626,240 @@ pub fn search_smb_record(i: &[u8]) -> IResult<&[u8], &[u8]> {
...
+        if guid.len() == 16 {
+            let output = format!("{:02x}{:02x}{:02x}{:02x}-{:02x}{:02x}-{:02x}{:02x}-{:02x}{:02x}-{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}",
+                    guid[3],  guid[2],  guid[1],  guid[0],
+                    guid[5],  guid[4],  guid[7],  guid[6],
+                    guid[9],  guid[8],  guid[11], guid[10],
+                    guid[15], guid[14], guid[13], guid[12]);
+            output
+        } else {
+            "".to_string()
+        }
```
[the remark this answers is not in the document]
Comment (b1tg) [reply rust/src/smb/smb2_records.rs:646]: I borrow this function from here, do I need change it too? https://github.com/OISF/suricata/blob/a94ca4462093c0b41f87a7d8433801a0abbb4390/rust/src/smb/log.rs#L55

[the remark this answers is not in the document]
Comment (inashivb) [reply rust/src/smb/smb2_records.rs:646]: Sure. In fact, is it possible to only change it in this `log.rs` fn and `use` that here? Haven't tested myself but if it is possible we could avoid duplication of the fn here.

**Comment by inashivb** [comment]
> > Could you please add the source of test data to every test?
> 
> Is this style suitable? @inashivb
> 
> https://github.com/OISF/suricata/blob/af55721842d20f1d3ae70aef6144428cf7cda709/rust/src/smb/smb2_records.rs#L686-L689

This looks fine to me as it is informative enough. But, you'll hear from others if they have a different opinion.
Thank you for your continued interest and work! :)

**Comment by victorjulien** [comment]
Merged in #8815, thanks!

Did some clippy fixes and reformatting on top of this.

**Pull request closed without merging**